bin/mztab_output.py

```diff
@@ -0,0 +1,153 @@
+
+    df=df[['sequence', 'RT', 'charge', 'Sample', 'intensity']]
+    df.columns=['PeptideSequence', 'retention_time', 'PrecursorCharge', 'Run', 'Intensity']
+
+    df=df.merge(df_lib, how='inner', on='PeptideSequence')
+    df=df.drop_duplicates()
+
+    #reformat as PRT protein mzTab
+    remaining_prh_cols='unit_id,description,taxid,species,database,database_version,search_engine,search_engine_score,reliability,num_peptides_unambiguous,ambiguity_members,modifications,uri,go_terms,protein_coverage'.split(',')
+
+    num_peptides=df.groupby(['ProteinName'])['PeptideSequence'].apply(list).apply(len).reset_index()['PeptideSequence']
+    num_peptides_distinct=df.groupby(['ProteinName'])['PeptideSequence'].apply(list).apply(set).apply(len).reset_index()['PeptideSequence']
+    df_prot=df.groupby(['ProteinName','Run'])['Intensity'].apply(sum).reset_index().pivot(index='ProteinName', columns='Run', values='Intensity').reset_index()
+    col_idxs=df_prot.columns[1:].tolist()
+    col_idxs_enum=[i[0]+1 for i in enumerate(col_idxs)]
+    df_prot.columns=['accession']+['protein_abundance_sub['+str(i)+']' for i in col_idxs_enum]
+    for i in col_idxs_enum:
+        df_prot['protein_abundance_stdev_sub['+str(i)+']']='null'
+        df_prot['protein_abundance_std_error_sub['+str(i)+']']='null'
+
+    cols=df_prot.columns.tolist()
+    df_prot['PRH']='PRT'
+    df_prot=df_prot[['PRH']+cols]
+    df_prot['num_peptides']=num_peptides
+    df_prot['num_peptides_distinct']=num_peptides_distinct
+    for c in remaining_prh_cols:
+        df_prot[c]='null'
+
+    df_prot.to_csv('PRT_dataframe.csv',index=False, sep='\t')
+    op=open('PRT_dataframe.csv','r')
+    opr_prt=op.readlines()
+    op.close()
+
+    #reformat as PEP peptide mzTab
+    remaining_pep_cols='unit_id,unique,database,database_version,search_engine_score,reliability,modifications,uri,spectra_ref'.split(',')
+
+    retention_time=df.groupby(['PeptideSequence','ProteinName','PrecursorCharge','PrecursorMz'])['retention_time'].apply(np.median).reset_index()['retention_time']
+    df_pep=df.groupby(['PeptideSequence','ProteinName','PrecursorCharge','PrecursorMz','Run'])['Intensity'].apply(sum).unstack('Run').reset_index()
+    df_pep.columns=['sequence','accession','charge','mass_to_charge']+['peptide_abundance_sub['+str(i)+']' for i in col_idxs_enum]
+    for i in col_idxs_enum:
+        df_pep['peptide_abundance_stdev_sub['+str(i)+']']='null'
+        df_pep['peptide_abundance_std_error_sub['+str(i)+']']='null'
+
+    cols=df_pep.columns.tolist()
+    df_pep['PEH']='PEP'
+    df_pep=df_pep[['PEH']+cols]
+    df_pep['retention_time']=retention_time
+    df_pep['search_engine']='OpenSwathWorkflow'
+    for c in remaining_pep_cols:
+        df_pep[c]='null'
+
+    df_pep.to_csv('PEP_dataframe.csv',index=False, sep='\t')
+
+    op=open('PEP_dataframe.csv','r')
+    opr_pep=op.readlines()
+    op.close()
+
+    #reformat MTD header
+    header=['\t'.join(['MTD','mzTab-version','1.0'])+'\n']
```

Comment:
I think there are some more mandatory meta values that you could easily add. See e.g. https://psidev.info/sites/default/files/2017-07/v_1_0_0_20minute_guide_mzTab.pdf
mzTab-mode etc. can be easily hard coded

Reply:
I'll have another look what else would be easy to incorporate

Reply:
done